db/migrate/20200129172534_add_epoch_version_release_arch_to_katello_installed_packages.rb

```diff
@@ -0,0 +1,38 @@
+class AddEpochVersionReleaseArchToKatelloInstalledPackages < ActiveRecord::Migration[5.2]
+  def up
+    add_column :katello_installed_packages, :nvrea, :string
+    add_column :katello_installed_packages, :epoch, :string
+    add_column :katello_installed_packages, :version, :string
+    add_column :katello_installed_packages, :release, :string
+    add_column :katello_installed_packages, :arch, :string
+
+    epoch_non_0 = ::Katello::Rpm.where.not(epoch: [0, nil]).pluck(:nvra, :epoch)
+    installed_packages = []
+
+    ::Katello::InstalledPackage.reset_column_information
+    ::Katello::InstalledPackage.find_each do |pkg|
+      #epoch = epoch_non_0.select{ |nvra_epoch| nvra_epoch[0] == pkg.nvra }.first&.second || "0"
+      epoch_non_0_map = epoch_non_0.map.to_h
```

Comment:
you can move this out of the loop to only do it once?  Also don't need the previous line